core/contracts/financial-templates/implementation/demo/DepositBox.sol

```diff
@@ -0,0 +1,346 @@
+            depositBoxData.requestPassTimestamp != 0 && depositBoxData.requestPassTimestamp <= getCurrentTime(),
+            "Invalid withdraw request"
+        );
+
+        // Get the resolved price or revert.
+        FixedPoint.Unsigned memory exchangeRate = _getOraclePrice(depositBoxData.requestPassTimestamp);
+
+        // Calculate denomated amount of collateral based on resolved exchange rate.
+        // Example 1: User wants to withdraw $100 of ETH, exchange rate is $200/ETH, therefore user to receive 0.5 ETH.
+        // Example 2: User wants to withdraw $250 of ETH, exchange rate is $200/ETH, therefore user to receive 1.25 ETH.
+        FixedPoint.Unsigned memory denominatedAmountToWithdraw = depositBoxData.withdrawalRequestAmount.div(
+            exchangeRate
+        );
+
+        // If withdrawal request amount is > collateral, then withdraw the full collateral amount and delete the deposit box data.
+        if (denominatedAmountToWithdraw.isGreaterThan(_getFeeAdjustedCollateral(depositBoxData.rawCollateral))) {
+            denominatedAmountToWithdraw = _getFeeAdjustedCollateral(depositBoxData.rawCollateral);
+
+            // Reset the position state as all the value has been removed after settlement.
+            emit EndedDepositBox(msg.sender);
+        }
+
+        // Decrease the individual deposit box and global collateral balance.
+        amountWithdrawn = _decrementCollateralBalances(depositBoxData, denominatedAmountToWithdraw);
+
+        emit RequestWithdrawalExecuted(
+            msg.sender,
+            amountWithdrawn.rawValue,
+            exchangeRate.rawValue,
+            depositBoxData.requestPassTimestamp
+        );
+
+        // Reset withdrawal request by setting withdrawal request timestamp to 0.
+        _resetWithdrawalRequest(depositBoxData);
+
+        // Transfer approved withdrawal amount from the contract to the caller.
+        collateralCurrency.safeTransfer(msg.sender, amountWithdrawn.rawValue);
+    }
+
+    /**
+     * @notice Cancels a pending withdrawal request.
+     */
+    function cancelWithdrawal() external nonReentrant() {
+        DepositBoxData storage depositBoxData = depositBoxes[msg.sender];
+        require(depositBoxData.requestPassTimestamp != 0, "No pending withdrawal");
+
+        emit RequestWithdrawalCanceled(
+            msg.sender,
+            depositBoxData.withdrawalRequestAmount.rawValue,
+            depositBoxData.requestPassTimestamp
+        );
+
+        // Reset withdrawal request by setting withdrawal request timestamp to 0.
+        _resetWithdrawalRequest(depositBoxData);
+    }
+
+    /**
+     * @notice `emergencyShutdown` and `remargin` are required to be implemented by all financial contracts and exposed to the DVM, but
+     * because this is a minimal demo they will simply exit silently.
+     */
+    function emergencyShutdown() external override nonReentrant() {
+        return;
+    }
+
+    function remargin() external override nonReentrant() {
```

Comment:
do we have to have this function? if so prehaps consider adding a comment?

Reply:
I'm pretty sure it must override these two functions if it inherits `AdminstrateeInterface` and I think we require all financial templates to inherit this, right @mrice32 ?